### DISCUSSION TITLE
JoinError::Cancelled being returned without `abort` ever being called

### DISCUSSION THREAD

**Discussion by dr-bonez** [tokio-rs/tokio#5345] [Q&A] [answered]
Is it possible for the tokio runtime to cancel a task without `abort` ever being called?

I have a task created by `tokio::spawn` that is stored, but never referenced anywhere. In order to figure out why data in the task is being dropped, I added a `dbg!(futures::poll!(&mut self._task));` and got `JoinError::Cancelled`. How is this possible if `.abort()` is never called on the handle? The task is not being passed to any libraries (like I said, not referenced *at all* besides that `dbg!`) and the runtime is not being shutdown.

**Comment by Darksonn** [comment, marked as answer]
It can happen when you shut down the runtime. Otherwise, it is not possible.

**Reply by dr-bonez** [reply]
Yeah, I'm an idiot 🤦🏽 

This future was being created on a different runtime than it was being polled on due to a lazy init. It was a hack we did to be able to run non-Send futures. I just needed to make sure that it was being created on the correct runtime by passing the runtime to the lazy initializer.
